### ISSUE TITLE
Project is not being processed if its window is hidden by other windows

### ISSUE THREAD

**Issue by SymmetryUnbroken** [godotengine/godot#57683]
### Godot version

3.4.2 stable

### System information

Ubuntu 20.04.3 LTS, GLES3

### Issue description

Project is not being processed if its window is obscured by other windows. My project is extremely simple: a timer, and a text label with a number which the timer increments every second. However, if I alt-tab to another window (for example, a browser), and expect the number to tick in the background, it does not happen. I can wait for about 20 seconds with window not in focus, but the number in the label would advance by 1-2 seconds only. The issue only arises if the game window is not minimised, but obscured by other windows. If the project window is minimised instead, the slowdown does not occur.

### Steps to reproduce

1) Run the minimal reproduction project
2) Observe the number incrementing
3) Remember the current number and alt-tab to some other window, e.g. a browser, so that the game window is completely obscured
4) Wait for 20-30 seconds
5) Alt-tab back, observe the number has stayed almost the same

### Minimal reproduction project

[Lag showcase.zip](https://github.com/godotengine/godot/files/8009107/Lag.showcase.zip)

**Comment by Calinou** [comment]
@SymmetryUnbroken Can you reproduce this on [4.0 alpha1](https://godotengine.org/article/dev-snapshot-godot-4-0-alpha-1)?

**Comment by SymmetryUnbroken** [comment]
@Calinou Yes, I can. (Though I have to change line 6 of the script to ```$Timer.connect("timeout", _on_Timer_timeout)``` , presumably because syntax of connect was changed)
